client/hwlib/debian/rules

```diff
@@ -1,47 +1,9 @@
 #!/usr/bin/make -f
 
-include /usr/share/dpkg/pkg-info.mk
-include /usr/share/dpkg/architecture.mk
-include /usr/share/dpkg/buildflags.mk
-include /usr/share/rustc/architecture.mk
-export CFLAGS CXXFLAGS CPPFLAGS LDFLAGS
-export DEB_HOST_RUST_TYPE DEB_HOST_GNU_TYPE
-export PATH := /usr/share/cargo/bin:$(PATH)
-export CARGO=/usr/share/cargo/bin/cargo
-export CARGO_HOME=$(CURDIR)/debian/cargo_home
-export DEB_CARGO_CRATE=$(DEB_SOURCE)_$(DEB_VERSION_UPSTREAM)
-export DEB_BUILD_MAINT_OPTIONS=hardering=+bindnow
+export CARGO_HOME=$(CURDIR)/debian/
 
 %:
-	dh $@
+	dh $@ --buildsystem cargo
 
-override_dh_auto_clean:
-	dh_auto_clean
-	cargo clean
-	rm -rf target/release
-	rm -rf debian/cargo_registry
-
-override_dh_auto_configure:
-	$(CARGO) prepare-debian debian/cargo_registry --link-from-system
-	rm -f Cargo.lock
-	dh_auto_configure
-
-override_dh_auto_build:
-	dh_auto_build
-	# Build the Rust library
-	mkdir -p target
-	CARGO_TARGET_DIR=./target $(CARGO) build --release --lib --offline
-
-override_dh_auto_install:
-	dh_auto_install
-	# Install the Rust library
-	mkdir -p debian/hwlib/usr/lib/$(DEB_HOST_MULTIARCH)/
-	mkdir -p debian/hwlib/usr/lib/python3/dist-packages/
-	install -Dm755 target/$(DEB_HOST_RUST_TYPE)/release/libhwlib.so debian/hwlib/usr/lib/$(DEB_HOST_MULTIARCH)/
-	# Install the Python bindings generated by maturin
-	python3 -m pip install ./*.whl --no-deps --target debian/hwlib/usr/lib/python3/dist-packages/ --no-cache-dir
-
-override_dh_shlibdeps:
-	dh_shlibdeps -l$(CURDIR)/debian/hwlib/usr/lib/python3/dist-packages/hwlib.libs
-
-override_dh_dwz:
+override_dh_auto_test:
+	/usr/share/cargo/bin/cargo test -- --test-threads=1
```

Comment:
Nice reduction!

Why do we need --test-threads=1? Please add a comment for the next maintainer.

Reply:
This is how we run the tests in general for hwlib even in CI.
We do it this way because our tests (for now, will be changed with [this task](https://warthogs.atlassian.net/browse/C3-681)) rely on the environment variable value. So, `cargo test -- --test-threads=1` forces the tests to run sequentially, which avoids potential issues with environment variables being modified by concurrent tests

Reply:
You just said: the code is buggy.

Please add this comment to the repo code, file a ticket and link it there so that it can be removed.

Tests that modify environment variables need to grab a lock or use some other method. A single threaded runner is not a workaround.